knimin/lib/parse.py

```diff
@@ -34,3 +36,60 @@ def parse_plate_reader_output(contents):
         data.append(line.split())
 
     return np.asarray(data, dtype=np.float)
+
+
+def parse_qpcr_object(contents):
+    """ Parses a QPCR machine string, returning the per well Cps
+
+    Parameters
+    ----------
+    contents : str
+        The string generated by the QPCR machine to parse.
+        New lines \n, field \t
+
+    Returns
+    -------
+    numpy.array 2D
+        The per well concentrations
+
+    Raises
+    ------
+        ValueError
+            If Cp or Pos is not one of the headers
+    """
+    contents = pd.read_csv(
+            StringIO(contents),
+            sep='\t',
+            dtype=str,
+            encoding='utf-8',
+            infer_datetime_format=False,
+            keep_default_na=False,
+            index_col=False,
+            comment='\t')
+
+    headers = contents.columns.values
+    if 'Cp' not in headers or 'Pos' not in headers:
+        raise ValueError("The 'Cp' and 'Pos' headers are "
+                         "required. The ones present are: %s" % headers)
+
+    # Pos format is A1 -> [0][0], B24 -> [1][23], etc so the max row will
+    # be the first letter - 'A' + 1
+    max_row = sorted(contents.Pos.values,
+                     key=lambda item: item[0])[-1][0]
+    max_row = ord(max_row) - ord('A') + 1
+    # the max columns is the max number after the letter - 1
+    max_col = sorted(contents.Pos.values,
+                     key=lambda item: int(item[1:]))[-1][1:]
+    max_col = int(max_col)
```

Comment:
turns out that google spreadsheets copy/pastes tabs even if the original file has commas ... changed in last pr.